bindings/src/main/java/io/pravega/storage/extendeds3/ExtendedS3ChunkStorage.java

```diff
@@ -295,6 +298,17 @@ protected void doDelete(ChunkHandle handle) throws ChunkStorageException {
         }
     }
 
+    @Override
+    public void close() {
+        if (!this.closed.getAndSet(true)) {
+            try {
+                this.client.destroy();
```

Comment:
While this is going to work, I don't like the pattern. This `client` is given to this class via the constructor so there is no guarantee that nobody else is using it. If you are closing it, then anyone else who might be using it will be in trouble.

Please refactor the factory class and this class to have this class construct the object - which means it owns it wholly.